### PR TITLE
Add monthly contributor rewards claim

### PR DESCRIPTION
**Pull request by ncitron**
Added a new section under the Resources section called Rewards where the monthly contributor rewards can be claimed. I will be making it a bit easier for us to add new Merkle Distributors for future rewards in the next few days, but I wanted to get this out ASAP so everyone can easily claim their rewards.

### CODE REVIEW

File: src/contexts/Rewards/Provider.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import ConfirmTransactionModal, {
+  TransactionStatusType,
+} from 'components/ConfirmTransactionModal'
+import useWallet from 'hooks/useWallet'
+import { claimRewards, getUnclaimedRewards } from 'index-sdk/rewards'
+import { waitTransaction } from 'utils/index'
+import {
+  checkIsAirdropClaimed,
+  getAirdropDataForAddress,
+} from '../../index-sdk'
```
Comment (controtie) [inline src/contexts/Rewards/Provider.tsx:15]: Delete

File: src/views/ContributorRewards/ContributorRewards.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import Page from '../../components/Page'
+import styled from 'styled-components'
+import { Container, Spacer } from 'react-neu'
+import MonthlyClaim from './components/MonthlyClaim'
```
Comment (controtie) [inline src/views/ContributorRewards/ContributorRewards.tsx:5]: order imports + use the shorthand for Page here
`import Page from 'components/Page'`

File: src/views/ContributorRewards/components/MonthlyClaim.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useMemo } from 'react'
+import { Button, Card, CardActions, CardContent, Spacer } from 'react-neu'
+import styled from 'styled-components'
+import useWallet from '../../../hooks/useWallet'
+import MonthsDropdown from './MonthsDropdown'
+import useRewards from '../../../hooks/useRewards'
```
Comment (controtie) [inline src/views/ContributorRewards/components/MonthlyClaim.tsx:6]: clean up

File: src/index-sdk/rewards.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+): Promise<BigNumber> => {
+  const merkleAccount = getMerkleAccount(account)
+  if (await isClaimed(provider, account)) {
+    return new BigNumber(0)
+  } else {
+    return new BigNumber(merkleAccount?.amount || 0).div(
+      new BigNumber(Math.pow(10, 18))
+    )
+  }
+}
```
Comment (controtie) [inline src/index-sdk/rewards.ts:77]: Two things:
I think it makes sense to make this implementation identical to `airdrop.ts`.
The airdrop and monthly contributor rewards are nearly identical in functionality. The airdrop existing airdrop implementation has the added benefit of being tested/in production for a while

Please add more line breaks in between your code, it makes it easier to read. For example on isClaimed above ^ there's a natural break in between "setting the stage" (51 & 52) and executing (54 - 62)

**Review by controtie — CHANGES_REQUESTED** [review]
Looks good for the most part. Requested some code style changes.
But this should be a dialogue, let me know if you have any disagreements

**Comment by ncitron** [comment]
Should be good to go now. Let me know if you want any additional changes.

File: src/index-sdk/rewards.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import { provider } from 'web3-core'
+import { AbiItem } from 'web3-utils'
+import MerkleABI from './abi/MerkleDistributor.json'
+import merkleData from './novemberRewardsMerkle.json'
+
+export const getMerkleContract = (provider: provider) => {
+  const web3 = new Web3(provider)
+  return new web3.eth.Contract(
+    (MerkleABI as unknown) as AbiItem,
+    process.env.REACT_APP_NOVEMBER_REWARDS_ADDRESS
```
Comment (controtie) [inline src/index-sdk/rewards.ts:11]: typically we import/export all eth contract addresses in `ethContractAddresses.ts`

File: src/index-sdk/rewards.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  )
+}
+
+const getMerkleAccount = (account: string) => {
+  let key = Object.keys(merkleData).find(
+    (key) => account.toLowerCase() === key.toLowerCase()
```
Comment (controtie) [inline src/index-sdk/rewards.ts:17]: `account.toLowerCase()` will break if account is undefined or null, which looks like a possibility.
Can add an optional chain to fix this.

**Review by controtie — APPROVED** [review]
Thanks for the changes, LGTM

**Pull request merged**